tanner/api/api.py

```diff
@@ -271,33 +270,41 @@ def check_time(time):
 
         tables = "sessions S"
         columns = "S.id"
-        where = "S.sensor_id='%s'" % (filters["sensor_id"])
-
-        if "attack_type" in filters:
-            tables += ", paths P"
-            columns += ", P.session_id"
-            try:
-                attack_type = AttackType[filters["attack_type"]].value
-            except KeyError:
-                return "Invalid filter value"
-
-            where += " AND P.attack_type=%s AND S.id=P.session_id" % (attack_type)
-        if "owners" in filters:
-            tables += ", owners O"
-            columns += ", O.session_id"
-            where += " AND O.owner_type='%s' AND S.id=O.session_id" % (
-                filters["owners"]
-            )
-        if "start_time" in filters:
-            start_time = check_time(filters["start_time"])
-            where += " AND S.start_time>='%s'" % (start_time)
-        if "end_time" in filters:
-            end_time = check_time(filters["end_time"])
-            where += " AND S.end_time<='%s'" % (end_time)
-        if "peer_ip" in filters:
-            where += " AND S.ip='%s'" % (filters["peer_ip"])
-        if "user_agent" in filters:
-            where += " AND S.user_agent='%s'" % (filters["user_agent"])
+        where = "S.sensor_id='%s'" % (filters["sensor_id"][0])
+
+        for parameter, values in filters.items():
+            if parameter == "attack_type":
```

Comment:
I thought a lot about this query building, can we rebuilt it in a way it looks more accurate and we use the power of sqlalchemy? like here https://aiopg.readthedocs.io/en/stable/examples.html#complex-sqlalchemy-queries

Reply:
I actually thought about it and I think almost everything would stay similar to the way it is. 
The only thing that would differ is that instead of having raw queries we would be using the sqlalchemy objects 

I mean all these conditions would have to stay so code might not become as clean as we think.

But I can definitely try to do it.

Reply:
We could keep it in mind and make as a separate task :)